src/partitioning/block.jl

```diff
@@ -54,17 +54,25 @@ function partition(object, method::BlockPartition)
 
     append!(subsets[i], j)
   end
+  
+  #Intitialize metadata to an empty Dict.
+  #If method.neighbors is enabled, we will populate it.
+  metadata = Dict()
 
   # neighboring blocks metadata
-  bstart  = CartesianIndex(ntuple(i -> 1, Dim))
-  boffset = CartesianIndex(ntuple(i -> 1, Dim))
-  bfinish = CartesianIndex(Dims(nblocks))
-  for (i, bcoords) in enumerate(bstart:bfinish)
-    for b in (bcoords - boffset):(bcoords + boffset)
-      if all(Tuple(bstart) .≤ Tuple(b) .≤ Tuple(bfinish)) && b ≠ bcoords
-        push!(neighbors[i], linear[b])
+  if method.neighbors == true
+    bstart  = CartesianIndex(ntuple(i -> 1, Dim))
+    boffset = CartesianIndex(ntuple(i -> 1, Dim))
+    bfinish = CartesianIndex(Dims(nblocks))
+    for (i, bcoords) in enumerate(bstart:bfinish)
+      for b in (bcoords - boffset):(bcoords + boffset)
+        if all(Tuple(bstart) .≤ Tuple(b) .≤ Tuple(bfinish)) && b ≠ bcoords
+          push!(neighbors[i], linear[b])
+        end
       end
     end
```

Comment:
Can you please double check the indentation? I find it strange that GitHub only shows changes for a part of the loop. Double check that the indentation is correct locally and is not using TAB

Reply:
Checking.

Reply:
The indentation seems to be correct locally. I have re-written it as well just to make sure.
![Screenshot from 2021-03-09 02-33-48](https://user-images.githubusercontent.com/14348863/110381899-24699300-8080-11eb-9780-63ed70fe9144.png)